### PR TITLE
Fixes jump boots sleeper bug, adds return value to /atom/movable/throw_at

### LINKED ISSUE #30963: Jump Boots break if used in a sleeper
Round # 22905 (Hippie)

Reproduced and confirmed by me, an administrator. If Jump boots are used in a sleeper, the HUD icon for the user no longer produces the desired effect.

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
Fixes #30963

Also, modifies /atom/movable/throw_at to return FALSE on an early failure condition (= callback will not be called) or TRUE otherwise (= callback will be called.)

Handling the return value will allow stuff that relies on a throw_at callback to continue functioning.

[Changelogs]: 
:cl: Naksu
fix: Fixed jump boots breaking if used when you can't jump, such as when inside a sleeper.
/:cl:

[why]: 
Bugfix

### CODE REVIEW

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 1]
```diff
@@ -207,10 +207,12 @@
 
 	var/atom/target = get_edge_target_turf(usr, usr.dir) //gets the user's direction
 
-	jumping = TRUE
-	playsound(src.loc, 'sound/effects/stealthoff.ogg', 50, 1, 1)
-	usr.visible_message("<span class='warning'>[usr] dashes forward into the air!</span>")
-	usr.throw_at(target, jumpdistance, jumpspeed, spin=0, diagonals_first = 1, callback = CALLBACK(src, .proc/hop_end))
+	if (usr.throw_at(target, jumpdistance, jumpspeed, spin=0, diagonals_first = 1, callback = CALLBACK(src, .proc/hop_end)))
+		jumping = TRUE
+		playsound(src.loc, 'sound/effects/stealthoff.ogg', 50, 1, 1)
```
Comment (Cyberboss) [inline code/modules/clothing/shoes/miscellaneous.dm]: remove .loc

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 2]
```diff
@@ -207,10 +207,12 @@
 
 	var/atom/target = get_edge_target_turf(usr, usr.dir) //gets the user's direction
 
-	jumping = TRUE
-	playsound(src.loc, 'sound/effects/stealthoff.ogg', 50, 1, 1)
-	usr.visible_message("<span class='warning'>[usr] dashes forward into the air!</span>")
-	usr.throw_at(target, jumpdistance, jumpspeed, spin=0, diagonals_first = 1, callback = CALLBACK(src, .proc/hop_end))
+	if (usr.throw_at(target, jumpdistance, jumpspeed, spin=0, diagonals_first = 1, callback = CALLBACK(src, .proc/hop_end)))
```
Comment (Cyberboss) [inline code/modules/clothing/shoes/miscellaneous.dm]: spin = FALSE, diagonals_first = TRUE

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — DISMISSED** [review]

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 3]
```diff
@@ -207,10 +207,12 @@
 
 	var/atom/target = get_edge_target_turf(usr, usr.dir) //gets the user's direction
 
-	jumping = TRUE
-	playsound(src.loc, 'sound/effects/stealthoff.ogg', 50, 1, 1)
-	usr.visible_message("<span class='warning'>[usr] dashes forward into the air!</span>")
-	usr.throw_at(target, jumpdistance, jumpspeed, spin=0, diagonals_first = 1, callback = CALLBACK(src, .proc/hop_end))
+	if (usr.throw_at(target, jumpdistance, jumpspeed, spin = TRUE, diagonals_first = TRUE, callback = CALLBACK(src, .proc/hop_end)))
```
Comment (optimumtact) [inline code/modules/clothing/shoes/miscellaneous.dm]: really, there's a fucking user arg passed into this proc

come on

Comment (Cyberboss) [inline code/modules/clothing/shoes/miscellaneous.dm]: usr -> user

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 4]
```diff
@@ -207,10 +207,12 @@
...
+		jumping = TRUE
+		playsound(src, 'sound/effects/stealthoff.ogg', 50, 1, 1)
+		usr.visible_message("<span class='warning'>[usr] dashes forward into the air!</span>")
```
Comment (Cyberboss) [inline code/modules/clothing/shoes/miscellaneous.dm]: usr -> user

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 5]
```diff
@@ -207,10 +207,12 @@
...
+	else
+		to_chat(usr, "<span class='warning'>Something prevents you from dashing forward!</span>")
```
Comment (Cyberboss) [inline code/modules/clothing/shoes/miscellaneous.dm]: usr -> user

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by vuonojenmustaturska** [comment]
Retested with the usr->user changes, also fixed a slight mistake no-one noticed

**Pull request merged**